src/transformers/models/fsmt/modeling_fsmt.py

```diff
@@ -530,6 +541,9 @@ def forward(
             if output_attentions:
                 all_attentions = all_attentions + (attn,)
 
+        if self.layer_norm is not None:
```

Comment:
for better readability would it be possible to continue with checking the `encoder_normalize_before` flag? As the currently suggested `if ...` doesn't tell the reader why this check is done, and why it may or may be defined.

and then you can drop this
```
        else:
            self.layer_norm = None
```
same for decoder.

Reply:
I agree with your review!

The original source code has been changed to the current one as follows to reflect the review.

- before review
```python
class FSMTEncoder(nn.Module):

    def __init__(self, config: FSMTConfig, embed_tokens: nn.Embedding):
        ...
        if config.encoder_normalize_before:
            self.layer_norm = nn.LayerNorm(embed_dim)
        else:
            self.layer_norm = None
        ...

    def forward(self, ...):
        ...
        if self.layer_norm is not None:
            x = self.layer_norm(x)
```
- after review
```python
class FSMTEncoder(nn.Module):

    def __init__(self, config: FSMTConfig, embed_tokens: nn.Embedding):
        ...
        self.pre_layernorm = config.encoder_pre_layernorm
        if self.pre_layernorm:
            self.layer_norm = nn.LayerNorm(embed_dim)
        ...

    def forward(self, ...):
        ...
        if self.pre_layernorm:
            x = self.layer_norm(x)
```